### PR TITLE
Refactor useSaveState and useSaveConflict to have less states

### PR DESCRIPTION
**Pull request by nsams**
depends on #748 

Also fix bug to show conflict dialog right after save

### CODE REVIEW

File: packages/admin/cms-admin/src/pages/createUsePage.tsx [hunk 1]
```diff
@@ -105,10 +104,7 @@ interface UsePageApi<PageState, RootBlocks extends RootBlocksInterface> {
     hasChanges?: boolean;
     loading: boolean;
     saving: boolean;
-    error: ApolloError | undefined;
-    saveError: ApolloError | undefined;
-    validating: boolean;
-    valid: boolean;
```
Comment (nsams) [inline packages/admin/cms-admin/src/pages/createUsePage.tsx:111]: incompatible change, but I don't think anyone used that?

File: packages/admin/cms-admin/src/common/useSaveState.tsx [hunk 2]
```diff
@@ -19,38 +19,37 @@ interface SaveStateOptions<TData> {
 interface SaveStateReturn {
     handleSaveClick: () => Promise<void>;
     saving: boolean;
-    saveError?: unknown;
-    valid: boolean;
-    validating: boolean;
```
Comment (nsams) [inline packages/admin/cms-admin/src/common/useSaveState.tsx:24]: incompatible change, but I don't think anyone used that?

**Review by johnnyomair — APPROVED** [review]

**Comment by johnnyomair** [comment]
I'm not sure why the `PageSaveButton` has `valid` and `validating` props. But I guess these states were only used internally

**Comment by nsams** [comment]
@johnnyomair conflicts fixed

**Pull request merged**